week3/src/App.jsx

```diff
@@ -0,0 +1,149 @@
+import styled, { ThemeProvider } from 'styled-components'
+import './style/App.css';
+import theme from './style/theme';
+import { useEffect, useState } from 'react';
+import { CARDS } from './cardData.jsx';
+
+function App() {
+  const [ success, setSuccess ] = useState(false);
+  const [ score, setScore ] = useState(0);
+
+  //filpedId의 값이 각 사진의 id == 배열에 id가 있으면 뒤집힌거 순서대로!!
+  const [ filpedId, setFilpedId ] = useState([]);
+  const [ prevOpenedId, setPrevOpenedId ] = useState(null);
+  
+  const [shuffledCards,setShuffledCards ] = useState([]);
+  useEffect(() => 
+    setShuffledCards([...CARDS].sort(()=> Math.random() - 0.5))
+    ,[success])
+
+  function handleFilp(item){
+    //여기에 item이 아니라 event로 보내서 했는데 그러니까 반영이 안되지!!!
+    //useState상태변경을 위해선 item자체를 보내야함!!!
+
+    //만약 filped 길이가 홀수개면 -> 하나 열린거 있느거임
+    //그럼 지금 열린게 같은 sort인지 확인후 추가
+    //만약 열린게 같은 sort가 아니라면 이 전거 pop,이번꺼 추가 안함
+    
+    //만약 짝수개라면 그냥 열기
+    //만약 길이가 13이라면 축하합니다 !
+
+    //전에 오픈했던데 틀렸다면 둘다 닫혀야 함
+    if(prevOpenedId !== null){
+      setFilpedId(
+        filpedId.filter( each => 
+          (each.sort !== prevOpenedId.sort)
+            &&(each !== filpedId[ filpedId.length -2 ])  )
+      )
+      setPrevOpenedId(null);
+    }
+    
+    const len = filpedId.length;
+
+    if( len % 2 !== 0 ){
+      //홀수
+      if( filpedId[len - 1].sort !== item.sort ){
+
+        //다른 종류!!!
+        setPrevOpenedId(item);
+        
+      } else{
+        setScore( score + 1 );
+      }
+    }
+
+    if( len >= 13){
+      setSuccess(true);
+    }
+  }
+  
+  function handleReTry(){
+    setFilpedId([]);
+    setPrevOpenedId(null);
+    setSuccess(false);
+    setScore(0);
+  }
+
+  return (
+  <ThemeProvider theme={theme}>
+    <header className='bigHeader'>
+      <h1>미니언즈 짝 맞추기</h1>
+      <Score>{`${score} / 7`}</Score>
+    </header>
```

Comment:
p5: 헤더 컴포넌트를 따로 분리하신 후에 src 폴더 아래 components 폴더를 만들고 그 안에 넣어서 사용하시는것도 추천드려요~  추후에 헤더에 기능이 추가 될 경우 재사용성을 높이실 수 있을거에요~

```suggestion
  <ThemeProvider theme={theme}>
   <Header score={score} />
```

Reply:
넵! 헤더 분리 했습니다 파일 구조도 추천해주신대로 만들었답니당 😄